modules/demo-magento/cypress/integration/checkout.js

```diff
@@ -61,4 +61,32 @@ describe('Checkout', () => {
 
     thankYouPage.isVisible()
   })
+
+  it('add valid coupon', () => {
+    product.visitRandom()
+    product.addToCart()
+    product.continueToCheckout()
+
+    checkout.getAddCoupon().click()
+    checkout.getCouponCode().type('coupon_cypress_test')
+    checkout.couponApply()
+    checkout.getAppliedCoupon().eq(1).should('have.text', 'coupon_cypress_test')
```

Comment:
What is eq(1)? Thats too specific to be inside it. If its here to ensure we get desktop applied coupon, move it to page object and rename the method to getAppliedCouponOnDesktop() or getDesktopAppliedCoupon() or something like that.